src/core/lv2/Lv2Proc.cpp

```diff
@@ -175,6 +176,26 @@ Lv2Proc::~Lv2Proc() { shutdownPlugin(); }
 
 
 
+void Lv2Proc::reload()
+{
+	// save controls, which we want to keep
+	QDomDocument doc;
+	QDomElement controls = doc.createElement("controls");
+	saveValues(doc, controls);
+	// backup construction variables
+	const LilvPlugin* plugin = m_plugin;
+	Model* parent = Model::parentModel();
+	// destroy everything using RAII ...
+	this->~Lv2Proc();
+	// ... and reuse it ("placement new")
+	new (this) Lv2Proc(plugin, parent);
```

Comment:
This isn't exception-safe: if `Lv2Proc`'s constructor throws an exception for some reason, then the object will be left in a destructed state, probably leading to a double-destruction later (which is undefined behaviour). The typical approach would be to construct a new `Lv2Proc`, then assign it to `*this`. Unfortunately, `QObject` is not assignable, but when we get Qt out of the core, we should revisit this.

Reply:
@DomClark Since the only one who calls `Lv2Proc::reload()` is `Lv2ControlBase`, which has an array of `Lv2Proc*`, couldn't we let `Lv2ControlBase` do the following?

1. save the contols + backup the construction variables
2. try to create the new Lv2Proc
3. try to load the values into the new Lv2Proc
4. only if 2+3 worked: do a pointer swap and delete the old Lv2Proc

Reply:
(Though this is worse than the current solution in that an additional heap allocation is required - but that's probably the negligible price of safety)